src/metabase/query_analysis.clj

```diff
@@ -62,10 +62,14 @@
   `(with-execution* ::disabled ~@body))
 
 (defn- execution
-  "The execution strategy for analysis, which can be overridden in dev and tests. In production, it is always async."
+  "The execution strategy for analysis. It can be overridden by a dynamic variable in dev and tests, and by a setting in production.
+
+  By default, it is async in production."
   []
   (case config/run-mode
-    :prod ::queued
+    :prod (if (public-settings/query-analysis-force-synchronicity)
+            ::immediate
+            ::queued)
```

Comment:
As discussed, if we're exposing this to customers I don't think its OK to override the behavior here. It will disable the rate limiting for the backfill and sweeping, and customers may opt-in to the P1 from v50.

Since we can avoid this sharp edge, I think that we should.